src/node/node_state.h

```diff
@@ -236,11 +237,12 @@ namespace ccf
       ringbuffer::AbstractWriterFactory& writer_factory,
       NetworkState& network,
       std::shared_ptr<enclave::RPCSessions> rpcsessions,
-      ShareManager& share_manager) :
+      ShareManager& share_manager,
+      const CurveID& curve_id) :
       sm(State::uninitialized),
       self(INVALID_ID),
-      node_sign_kp(tls::make_key_pair()),
-      node_encrypt_kp(tls::make_key_pair()),
+      node_sign_kp(tls::make_key_pair(curve_id)),
+      node_encrypt_kp(std::make_shared<tls::KeyPair_mbedTLS>(curve_id)),
```

Comment:
This still seems hardcoded to mbedTLS, as opposed to a generic compile-time configurable type?

Reply:
Yes, that's on purpose. This particular key pair is used in key_exchange, which we decided not to port to OpenSSL yet. It's pretty much the only thing left though - maybe I should take the time and port that too.